activestorage/lib/active_storage/attached/model.rb

```diff
@@ -189,6 +208,18 @@ def validate_service_configuration(association_name, service)
             end
           end
         end
+
+        def validate_key_interpolations(association_name, key)
+          return if key.blank?
+
+          key.split(KEY_PATH_SEPARATOR).map do |key_part|
+            key_part.scan(/:(\w*)/) do
+              unless ActiveStorage.key_interpolation_procs.keys.include?($1.to_sym)
+                raise ArgumentError, "Cannot configure #{key_part} in interpolation key '#{key}' for #{name}##{association_name}"
+              end
+            end
+          end
```

Comment:
You don't need to make a split here right? You can just use the scan method to pick the key_parts and then check if the interpolation key exists.
With this you won't need the `KEY_PATH_SEPARATOR` constant anymore here.

```suggestion
          key.scan(/:(\w+)/) do
            key_part = $1.to_sym
            unless ActiveStorage.key_interpolation_procs.keys.include?(key_part)
              raise ArgumentError, "Cannot configure #{key_part} in interpolation key '#{key}' for #{name}##{association_name}"
            end
          end
```